src/main/scala/org/idio/dbpedia/spotlight/CustomSpotlightModel.scala

```diff
@@ -521,6 +523,29 @@ class CustomSpotlightModel(val pathToFolder: String) {
 
   }
 
+
+  /*
+  * Change % of context vector
+  * */
+  def updatePercentageOfContextVector(surfaceForm: String, dbpediaURI: String, percentageOfVector: Double): Unit ={
+
+    try{
+      val surfaceFormId = this.customSurfaceFormStore.sfStore.getSurfaceForm(surfaceForm).id
+      val surfaceFormCounts = this.customSurfaceFormStore.sfStore.getSurfaceForm(surfaceForm).totalCount
+      val dbpediaId = this.customDbpediaResourceStore.resStore.getResourceByName(dbpediaURI).id
+      val surfaceFormAnnotatedCount = this.customSurfaceFormStore.sfStore.getSurfaceForm(surfaceForm).annotatedCount
+      this.customCandidateMapStore.changePercentageOfContextVector(surfaceFormAnnotatedCount, surfaceFormId, dbpediaId, surfaceFormCounts, percentageOfVector)
+
+    } catch {
+      case e: Exception => {
+        println("\t Given dbpediaURI or SF: " + dbpediaURI + " , " + surfaceForm + " could not be found")
```

Comment:
It would be good to have different messages for different errors. There are specific exceptions: `SurfaceFormNotFoundException` and `DBpediaResourceNotFoundException`.